src/server/game/Mails/ServerMailMgr.cpp

```diff
@@ -0,0 +1,331 @@
+            return;
+        }
+    } while (result->NextRow());
+
+    LoadMailServerTemplatesItems();
+    LoadMailServerTemplatesConditions();
+
+    LOG_INFO("server.loading", ">> Loaded {} Mail Server Template in {} ms", _serverMailStore.size(), GetMSTimeDiffToNow(oldMSTime));
+    LOG_INFO("server.loading", " ");
+}
+
+void ServerMailMgr::LoadMailServerTemplatesItems()
+{
+    QueryResult result = CharacterDatabase.Query("SELECT `templateID`, `faction`, `item`, `itemCount` FROM `mail_server_template_items`");
+    if (!result)
+    {
+        LOG_WARN("server.loading", ">> Loaded 0 server mail items. DB table `mail_server_template_items` is empty.");
+        return;
+    }
+
+    do
+    {
+        Field* fields = result->Fetch();
+
+        uint32 templateID = fields[0].Get<uint32>();
+        std::string faction = fields[1].Get<std::string>();
+        uint32 item = fields[2].Get<uint32>();
+        uint32 itemCount = fields[3].Get<uint32>();
+
+        if (_serverMailStore.find(templateID) == _serverMailStore.end())
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has an invalid templateID {}, skipped.", templateID);
+            continue;
+        }
+
+        ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(item);
+        if (!itemTemplate)
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has an invalid item {} for templateID {}, skipped.", item, templateID);
+            continue;
+        }
+
+        if (!itemCount)
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has itemCount 0 for item {}, skipped.", item);
+            continue;
+        }
+
+        uint32 stackable = itemTemplate->Stackable;
+        if (itemCount > stackable)
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has itemCount {} exceeding item_template.Stackable {} for item {}, skipped.", itemCount, stackable, item);
+            continue;
+        }
+
+        uint32 maxCount = itemTemplate->MaxCount;
+        if (maxCount && itemCount > maxCount)
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has itemCount {} exceeding item_template.MaxCount {} for item {}, skipped", itemCount, maxCount, item);
+            continue;
+        }
+
+        ServerMailItems mailItem;
+        mailItem.item = item;
+        mailItem.itemCount = itemCount;
+
+        if (faction == "Alliance")
+            _serverMailStore[templateID].itemsA.push_back(mailItem);
+        else if (faction == "Horde")
+            _serverMailStore[templateID].itemsH.push_back(mailItem);
+        else
+        {
+            LOG_ERROR("sql.sql", "Table `mail_server_template_items` has invalid faction value '{}' for templateID {}, skipped.", faction, templateID);
```

Comment:
Imo it's best to do the validation above where the rest of the validations are.

Reply:
I dont really understand? We need it after the other validation to know where to push it.
Also the `else` statement should in reality never be able to hit since we use enum in the database for faction